GameDesign/Assets/Scripts/MonoBehaviours/TimerManagerMonoBehaviour.cs

```diff
@@ -42,6 +55,28 @@ void Update()
         }
     }
 
+    private void checkWaves()
+    {
+        float[] wavePercentages = levelManager.waveMomentPercentages;
+        if(wavePercentages != null && currentWave < wavePercentages.Length)
+        {
+            float nextWaveTime = wavePercentages[currentWave] * maximumTime;
+            float timePassed = maximumTime - timeRemaining;
+            if (!isWaveAnnounced && (timePassed > (nextWaveTime - waveAnticipationTime)))
+            {
+                isWaveAnnounced = true;

```

Comment:
does this allow us to trigger a wave whenever we announce it? like if we want to force one?

Reply:
Every wave has an announcement event, in case we need to prepare for it. The announcement is given `waveAnticipationTime` seconds before the wave time.